### PR TITLE
Migrate NotificationChannelTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate NotificationChannelTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Akshay0701** [comment]
```
BUILD FAILED in 2m 52s
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:1: Error: Comment contents were corrupted by the Kotlin migration. For example: <http:></http:> [KotlinMigrationBrokenEmails from com.ichi2.anki:lint-rules]
/****************************************************************************************
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:83: Error: Use kotlin.test.assert[Not]Null OR kotlin.test.junit.JUnitAsserter instead of JUnit in Kotlin [LegacyNullAssertionDetector from com.ichi2.anki:lint-rules]
            assertNotNull(
            ^
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:44: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki:lint-rules]
    @get:Rule
    ^
3 errors, 0 warnings
```

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * Foundation; either version 3 of the License, or (at your option) any later           *
+ * version.                                                                             *
+ * *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY      *
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A      *
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.             *
+ * *
+ * You should have received a copy of the GNU General Public License along with         *
+ * this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.                           *
+ */
```
Comment (Akshay0701) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:15]: revert this change

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
@oyeraghib Thank for Contributions,
enter the proper name to your first commit

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt (hunk 1 above)
Comment (oyeraghib) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:15]: > @oyeraghib Thank for Contributions, enter the proper name to your first commit

should I name it as [Kotlin Migration] NotificationChannelTest.java ?

Comment (Akshay0701) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt]: convert the first commit name from `Rename .java to .kt` to `Rename NotificationChannelTest.java to NotificationChannelTest.kt`

Comment (david-allison) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt]: In future: we also have https://github.com/ankidroid/Anki-Android/blob/main/tools/migrate.sh which renames the commit

**Review by david-allison — CHANGES_REQUESTED** [review]
Fails lint

```
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.kt:84: Error: Use kotlin.test.assert[Not]Null OR kotlin.test.junit.JUnitAsserter instead of JUnit in Kotlin [LegacyNullAssertionDetector from com.ichi2.anki:lint-rules]
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
            assertNotNull(

            ^
```

**Comment by oyeraghib** [comment]
> 

not sure why but using "kotlin.test.junit.JUnitAsserter.assertNotNull" doesn't runs the test.

> e: D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\anki\tests\NotificationChannelTest.kt: (34, 15): Unresolved reference: test
e: D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\anki\tests\NotificationChannelTest.kt: (85, 13): Unresolved reference: assertNotNull
FAILURE: Build failed with an exception.

**Comment by david-allison** [comment]
Needs the following in `build.gradle`:


    androidTestImplementation "org.jetbrains.kotlin:kotlin-test:$kotlin_version"

**Comment by oyeraghib** [comment]
After adding the dependency and running the test it's showing few more deprecations in other files but they are just warnings but the test is still failing.
```
> Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac
D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\anki\DeckPickerTest.java:83: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
        assertThat(getActivityInstance(), instanceOf(StudyOptionsActivity.class));
        ^
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
error: warnings found and -Werror specified
D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\libanki\DBTest.java:46: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
        assertThat(journalMode.toLowerCase(Locale.ROOT), not(is("wal")));
        ^
  where T is a type-variable:
    T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
1 error
2 warnings

> Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac FAILED
> Task :AnkiDroid:deleteSingleFileInGradle UP-TO-DATE

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings
42 actionable tasks: 1 executed, 1 from cache, 40 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

```

**Comment by oyeraghib** [comment]
> After adding the dependency and running the test it's showing few more deprecations in other files but they are just warnings but the test is still failing.
> 
> ```
> > Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac
> D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\anki\DeckPickerTest.java:83: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
>         assertThat(getActivityInstance(), instanceOf(StudyOptionsActivity.class));
>         ^
>   where T is a type-variable:
>     T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
> error: warnings found and -Werror specified
> D:\anki\AnkiDroid\src\androidTest\java\com\ichi2\libanki\DBTest.java:46: warning: [deprecation] <T>assertThat(T,Matcher<? super T>) in Assert has been deprecated
>         assertThat(journalMode.toLowerCase(Locale.ROOT), not(is("wal")));
>         ^
>   where T is a type-variable:
>     T extends Object declared in method <T>assertThat(T,Matcher<? super T>)
> 1 error
> 2 warnings
> 
> > Task :AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac FAILED
> > Task :AnkiDroid:deleteSingleFileInGradle UP-TO-DATE
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
> 
> You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
> 
> See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings
> 42 actionable tasks: 1 executed, 1 from cache, 40 up-to-date
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':AnkiDroid:compileAmazonDebugAndroidTestJavaWithJavac'.
> > Compilation failed; see the compiler error output for details.
> 
> * Try:
> > Run with --stacktrace option to get the stack trace.
> > Run with --info or --debug option to get more log output.
> > Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 11s
> ```

Just adding the `import kotlin.test.junit.JUnitAsserter.assertNotNull` to build.gradle is making the tests failed.

**Comment by oyeraghib** [comment]
Just adding the `androidTestImplementation "org.jetbrains.kotlin:kotlin-test:$kotlin_version"
` in build.gradle is making the test failed. Even though I have not even imported `kotlin.test.junit.JUnitAsserter.assertNotNull` in the project yet.

**Pull request closed without merging**